### PR TITLE
Introduce Numerics class and Refactor Clamp Utils

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

There's a lot of cosmetic changes here but very little functional. I've been wanting to do a major cleanup for a while.

- Introduce `Numerics` utility method to consolidate non-image specific maths magritings several from `ImageMaths`.
- Replaces `Clamp` extension methods with static `Numerics` impl
- Migrates `Vector4` `FastClamp` to `Numerics`.
- Added vectorized `Clamp(Span, min, max)` methods (_As yet unused but have plans_).
- General renaming for consistency.

### CODE REVIEW

File: src/ImageSharp/Color/Color.cs [hunk 1]
```diff
@@ -27,19 +27,19 @@ namespace SixLabors.ImageSharp
         private Color(byte r, byte g, byte b, byte a)
         {
             this.data = new Rgba64(
-                ImageMaths.UpscaleFrom8BitTo16Bit(r),
-                ImageMaths.UpscaleFrom8BitTo16Bit(g),
-                ImageMaths.UpscaleFrom8BitTo16Bit(b),
-                ImageMaths.UpscaleFrom8BitTo16Bit(a));
+                ImageMath.UpscaleFrom8BitTo16Bit(r),
```
Comment (antonfirsov) [inline src/ImageSharp/Color/Color.cs:30]: I would move this to `Numerics` too. Or to a separate `ColorNumerics` (or similarly named) class.

File: src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ internal static class DenseMatrixUtils
             ref Vector4 target = ref Unsafe.Add(ref targetRowRef, column);
             vector.W = target.W;
 
-            Vector4Utilities.UnPremultiply(ref vector);
+            Vector4Utils.UnPremultiply(ref vector);
```
Comment (antonfirsov) [inline src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs:62]: `Numerics` or `ColorNumerics` ?

File: src/ImageSharp/Common/Helpers/Numerics.cs [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+
+                return adjustedCount;
+            }
+
+            return 0;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void ClampImpl<T>(Span<T> span, T min, T max)
+        where T : unmanaged
```
Comment (antonfirsov) [inline src/ImageSharp/Common/Helpers/Numerics.cs:396]:
```suggestion
            where T : unmanaged
```

File: src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs [hunk 4]
```diff
@@ -30,7 +30,7 @@ public SkewProcessor(float degreesX, float degreesY, Size sourceSize)
         /// <param name="sourceSize">The source image size</param>
         public SkewProcessor(float degreesX, float degreesY, IResampler sampler, Size sourceSize)
             : this(
-                 TransformUtilities.CreateSkewMatrixDegrees(degreesX, degreesY, sourceSize),
+                 TransformUtils.CreateSkewMatrixDegrees(degreesX, degreesY, sourceSize),
```
Comment (antonfirsov) [inline src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs:33]: So we are back to the `Utils` postfix? Agreed, shorter is better here.

**Review by antonfirsov — APPROVED** [review]
LGTM with suggestion:
We may want to move all Color/Pixel related numeric operations from `ImageMath` and `Vector4Utils` to a separate `ColorNumerics` class and probably get rid of `Vector4Utils` as a result. Cohesion around the `Vector4` type itself is no good cohesion here.

File: src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs (hunk 4 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs:33]: Yeah less typing. Unfortunately there's a public `GeometryUtilities` class I couldn't change.

File: src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs (hunk 2 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Common/Helpers/DenseMatrixUtils.cs:62]: I'll move both to `Numerics`.

File: src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs (hunk 4 above)
Comment (antonfirsov) [reply src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs:33]: We probably shouldn't have left that public for the 2 methods it has.

Comment (JimBobSquarePants) [reply src/ImageSharp/Processing/Processors/Transforms/Linear/SkewProcessor.cs:33]: Yeah... We shouldn't. Annoyed at myself for not spotting it.

**Review by antonfirsov — APPROVED** [review]

**Pull request merged**